dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeer.cpp

```diff
@@ -390,54 +395,66 @@ FastAddPeerRes FastAddPeerImplWrite(
     SYNC_FOR("in_FastAddPeerImplWrite::after_write_segments");
     if (cancel_handle->isCanceled())
     {
-        LOG_INFO(
-            log,
-            "FAP is canceled after write segments, region_id={} keyspace={} table_id={}",
-            region_id,
-            keyspace_id,
-            table_id);
+        LOG_INFO(log, "FAP is canceled after write segments");
         fap_ctx->cleanTask(tmt, proxy_helper, region_id, CheckpointIngestInfo::CleanReason::TiFlashCancel);
         GET_METRIC(tiflash_fap_task_result, type_failed_cancel).Increment();
         return genFastAddPeerResFail(FastAddPeerStatus::Canceled);
     }
 
-    // Write raft log to uni ps, we do this here because we store raft log seperately.
-    // Currently, FAP only handle when the peer is newly created in this store.
-    // TODO(fap) However, Move this to `ApplyFapSnapshot` and clean stale data, if FAP can later handle all snapshots.
+    // Now, the FAP snapshot is persisted. And we will later send it to ourselves to have is acked by the raftstore.
+    // Later, we directly write the raft log into formal unips. We do this here because we store raft log seperately.
+
+    // TODO(fap) Currently, FAP only handle when the peer is newly created in this store,
+    // Move this to `ApplyFapSnapshot` and clean stale data, if FAP can later handle all snapshots.
     UniversalWriteBatch wb;
     RUNTIME_CHECK(checkpoint_info->temp_ps != nullptr);
     RaftDataReader raft_data_reader(*(checkpoint_info->temp_ps));
+
+    bool too_many_raft_logs = false;
+    size_t log_count = 0;
     raft_data_reader.traverseRemoteRaftLogForRegion(
         region_id,
+        [&](size_t raft_log_count) {
+            GET_METRIC(tiflash_raft_raft_log_gap_count, type_unhandled_fap_raft_log).Observe(raft_log_count);
+            log_count = raft_log_count;
+            if (raft_log_count > 1500)
+            {
+                too_many_raft_logs = true;
+                // Will early abort.
+                return false;
+            }
+            return true;
+        },
         [&](const UniversalPageId & page_id, PageSize size, const PS::V3::CheckpointLocation & location) {
-            LOG_DEBUG(
-                log,
-                "Write raft log size {}, region_id={} index={}",
-                size,
-                region_id,
-                UniversalPageIdFormat::getU64ID(page_id));
+            LOG_DEBUG(log, "Write raft log size {} index={}", size, UniversalPageIdFormat::getU64ID(page_id));
```

Comment:
This may cause too many logging under debug level. I'd rather remove it.

Reply:
fixed

Reply:
the change is not yet pushed